### PR TITLE
OpenSSL Engine support

### PR DESCRIPTION
**Pull request by CIPop**
Adding OpenSSL Engine support.

### CODE REVIEW

File: iothub_client/tests/iothubclient_openssl_engine_e2e/Testing/Temporary/CTestCostData.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+---
```
Comment (danewalton) [inline iothub_client/tests/iothubclient_openssl_engine_e2e/Testing/Temporary/CTestCostData.txt:1]: this supposed to be included here?

**Review by danewalton — APPROVED** [review]

File: iothub_client/tests/iothubclient_openssl_engine_e2e/Testing/Temporary/CTestCostData.txt (hunk 1 above)
Comment (CIPop) [reply iothub_client/tests/iothubclient_openssl_engine_e2e/Testing/Temporary/CTestCostData.txt:1]: Wow: no. How did that happen - we might be lacking some .gitignore statements. I'll add one to ignore /Testing created by ctest.

File: iothub_client/samples/iothub_ll_client_x509_sample/iothub_ll_client_x509_sample.c [hunk 2]
```diff
@@ -75,6 +78,11 @@ static const char* x509privatekey =
 "xyJyxc700uV1eKiCdRLRuCbUeecOSZreh8YRIQQXoG8uotO5IttdVRc=""\n"
 "-----END RSA PRIVATE KEY-----";
 
+#ifdef SAMPLE_OPENSSL_ENGINE
+static const char* opensslEngine = SAMPLE_OPENSSL_ENGINE;
+static const OPTION_OPENSSL_KEY_TYPE x509keyfromengine = KEY_TYPE_ENGINE;
```
Comment (ericwol-msft) [inline iothub_client/samples/iothub_ll_client_x509_sample/iothub_ll_client_x509_sample.c:83]:
```suggestion
static const OPTION_OPENSSL_KEY_TYPE x509_key_from_engine = KEY_TYPE_ENGINE;
```

File: iothub_client/tests/iothubclient_openssl_engine_e2e/iothubclient_openssl_engine_e2e.c [hunk 3]
```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (ericwol-msft) [inline iothub_client/tests/iothubclient_openssl_engine_e2e/iothubclient_openssl_engine_e2e.c:1]: rename this to iothubclient_openssl_pks11_e2e.c ?

**Review by ericwol-msft — APPROVED** [review]

**Comment by CIPop** [comment]
Raspberry pi fails with setup issues. (@jbobotek PTAL)

**Pull request merged**